### PR TITLE
Support Go Modules

### PR DESCRIPTION
**Pull request by SamWhited**
This PR updates the method of dependency management to use Go Modules. Doing so means that no external tooling is required to update the vendor folder and manage dependencies. Since all supported versions of Go (and several before that) support Go Modules now, this PR should simplify the build for anyone using this plugin.

### CODE REVIEW

**Comment by tsandall** [comment]
@SamWhited thanks for submitting this patch. One issue that I noticed is that `$OPA_VERSION` does not appear to be set when the build runs. For example:

```
$ make
Building opa-docker-authz version: 0.5 (OPA version: )

Building opa-docker-authz ...

... done!
```

There was another PR submitted that updated to use v0.18.0, newer Go, etc. I've merged that because it didn't have any issues. That said, it would be nice to move to Go modules so if you can rebase/resolve the conflicts, we'll happily accept this.

**Comment by SamWhited** [comment]
I've fixed the merge conflict. As for the OPA version not being set, I don't seem to be able to reproduce it. What version of Go are you using, and what is the value of `GO111MODULE`? Maybe if you're using `GO111MODULE=auto` (or a version where that's the default) it can't do the lookup. Setting `GO111MODULE=on` for just that line could fix this.

**Comment by SamWhited** [comment]
I tried setting `GO111MODULE=on`, out of curiosity, see if that works for you.

**Comment by tsandall** [comment]
Hrmmm, what version of Make are you using? I've got GNU Make 3.81 on my macbook. I'm not sure about the OPA_VERSION declaration syntax you used. I made the following change and it fixed the issue:

```
$ git diff
diff --git a/Makefile b/Makefile
index 596e58f..bea2b3c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 .PHONY: all build

 VERSION := 0.6
-OPA_VERSION != GO111MODULE=on go list -m -f '{{.Version}}' github.com/open-policy-agent/opa
+OPA_VERSION := $(shell GO111MODULE=on go list -m -f '{{.Version}}' github.com/open-policy-agent/opa)
 GO_VERSION := 1.14.1
 REPO := openpolicyagent/opa-docker-authz
 DOCKER_VERSION := $(shell docker version --format '{{.Server.Version}}')
```

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+require (
+	github.com/Microsoft/go-winio v0.4.14 // indirect
+	github.com/OneOfOne/xxhash v1.2.5 // indirect
+	github.com/coreos/go-systemd v0.0.0-20190719114852-fd7a80b32e1f // indirect
+	github.com/docker/go-connections v0.4.1-0.20190612165340-fd1b1942c4d5 // indirect
+	github.com/docker/go-plugins-helpers v0.0.0-20170919092928-bd8c600f0cdd
+	github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32 // indirect
+	github.com/gobwas/glob v0.2.3 // indirect
+	github.com/open-policy-agent/opa v0.13.5
```
Comment (tsandall) [inline go.mod:13]: Use v0.18.0 here.

**Comment by SamWhited** [comment]
Oh wow, I see, so as it turns out that version of GNU Make is *very* old. The reason I wanted to use `!=` is that I use the version of Make from NetBSD. It tries to be more standards compliant than GNUMake does, and `!=` is a more portable syntax for assignment with shell expansion (the `$(shell)` stuff is a GNU-ism that nothing else supports).

The standard syntax has been supported by GNU Make since 2011 ([changelog](https://git.savannah.gnu.org/cgit/make.git/tree/ChangeLog.3#n1473)), however, the version of GNU Make you have installed is from *2006* ([changelog](https://git.savannah.gnu.org/cgit/make.git/tree/ChangeLog.3#n2881)).

May I ask what system you're using that still has a GNU Make version that old, or is it possible there's a newer version you could install? I'm not one of those people that tries to add features that only work on the bleeding edge, but I would like to build this with `bmake(1)` eventually which comes from NetBSD which is perfectly compatible with newer (but still quite old) versions of GNU Make. Thanks for your consideration.

File: go.mod (hunk 1 above)
Comment (SamWhited) [reply go.mod:13]: Done

**Comment by tsandall** [comment]
I'm running the latest macOS version (10.15.4) which ships with this ancient version of `make`. Since I ran into this, I'm guessing others would as well.

Since the `OPA_VERSION` variable is only used by the `build` target, could you just move the command into build.sh and set OPA_VERSION accordingly in there? That should be fine IMO.

**Comment by SamWhited** [comment]
> Since the OPA_VERSION variable is only used by the build target, could you just move the command into build.sh and set OPA_VERSION accordingly in there? That should be fine IMO.

Done

**Pull request merged**